src/ui/public/directives/input_base64.js

```diff
@@ -0,0 +1,81 @@
+import { uiModules } from 'ui/modules';
+import { Observable } from 'rxjs/Rx';
+const module = uiModules.get('kibana');
+
+const $readFileContents = (file) => {
+  return Observable.create(observer => {
+    const reader = new FileReader();
+    reader.onerror = (err) => {
+      observer.error(err);
+      observer.complete();
```

Comment:
`observer.error()` is like rejecting a promise, calling `observer.complete()` afterward does nothing.

Reply:
Good point, I'll remove the unnecessary completion.